File: src/uk/colessoft/android/hilllist/objects/ScootXMLHandler.java
package uk.colessoft.android.hilllist.objects;

import java.util.ArrayList;
import java.util.List;

import org.xml.sax.Attributes;
import org.xml.sax.SAXException;
import org.xml.sax.helpers.DefaultHandler;

public class ScootXMLHandler extends DefaultHandler {
	StringBuffer buff = null;
	ScootBusinesses scootBusinesses = new ScootBusinesses();
	Business tempBusiness;
	boolean buffering = false;
	ArrayList tempAddress = new ArrayList();

	@Override
	public void startDocument() throws SAXException {
		// Some sort of setting up work
	}

	@Override
	public void endDocument() throws SAXException {
		// Some sort of finishing up work
	}

	@Override
	public void startElement(String namespaceURI, String localName,
			String qName, Attributes atts) throws SAXException {
		if (localName.equals("companyname")) {
			buff = new StringBuffer("");
			buffering = true;
		} else if (localName.equals("longitude")) {
			buff = new StringBuffer("");
			buffering = true;
		} else if (localName.equals("latitude")) {
			buff = new StringBuffer("");
			buffering = true;
		} else if (localName.equals("result")) {
			buff = new StringBuffer("");
			buffering = true;
		} else if (localName.equals("telephone")) {
			buff = new StringBuffer("");
			buffering = true;
		} else if (localName.equals("postcode")) {
			buff = new StringBuffer("");
			buffering = true;
		} else if (localName.equals("linkback")) {
			buff = new StringBuffer("");
			buffering = true;
		} else if (localName.equals("l1address")) {

			buff = new StringBuffer("");
			buffering = true;
		} else if (localName.equals("l2address")) {

			buff = new StringBuffer("");
			buffering = true;
		} else if (localName.equals("l3address")) {

			buff = new StringBuffer("");
			buffering = true;
		} else if (localName.equals("l4address")) {

			buff = new StringBuffer("");
			buffering = true;
		} else if (localName.equals("l5address")) {

			buff = new StringBuffer("");
			buffering = true;
		} else if (localName.equals("l6address")) {

			buff = new StringBuffer("");
			buffering = true;
		}
	}

	@Override
	public void characters(char ch[], int start, int length) {
		if (buffering) {
			buff.append(ch, start, length);
		}
	}

	@Override
	public void endElement(String namespaceURI, String localName, String qName)
			throws SAXException {
		if (localName.equals("result")) {
			buffering = false;
			tempBusiness = new Business();
			tempBusiness.setResultNumber(Integer.parseInt(buff.toString()));
			if (this.getScootBusinesses().getBusinesses() == null)
				this.getScootBusinesses().setBusinesses(
						new ArrayList<Business>());
			this.scootBusinesses.getBusinesses().add(tempBusiness);
		} else if (localName.equals("companyname")) {
			buffering = false;
			tempBusiness.setCompanyname(buff.toString());

			// Do something with the full text content that we've just parsed
		} else if (localName.equals("longitude")) {
			buffering = false;
			tempBusiness.setLongitude(Float.parseFloat(buff.toString()));
		} else if (localName.equals("latitude")) {

			buffering = false;
			tempBusiness.setLatitude(Float.parseFloat(buff.toString()));
		} else if (localName.equals("telephone")) {

			buffering = false;
			tempBusiness.setTelephone(buff.toString());
		} else if (localName.equals("postcode")) {

			buffering = false;
			tempBusiness.setPostCode(buff.toString());
		} else if (localName.equals("linkback")) {

			buffering = false;
			tempBusiness.setScootLink(buff.toString());
		} else if (localName.equals("l1address")) {
			tempAddress = new ArrayList();
			buffering = false;
			tempAddress.add(buff.toString());

		} else if (localName.equals("l2address")) {
			buffering = false;
			tempAddress.add(buff.toString());
		} else if (localName.equals("l3address")) {
			buffering = false;
			tempAddress.add(buff.toString());
		} else if (localName.equals("l4address")) {
			buffering = false;
			tempAddress.add(buff.toString());
		} else if (localName.equals("l5address")) {
			buffering = false;
			tempAddress.add(buff.toString());
		} else if (localName.equals("l6address")) {
			buffering = false;
			tempAddress.add(buff.toString());
			tempBusiness.setAddress(tempAddress);
		}
	}

	public ScootBusinesses getScootBusinesses() {
		return scootBusinesses;
	}

	public void setScootBusinesses(ScootBusinesses scootBusinesses) {
		this.scootBusinesses = scootBusinesses;
	}
}

File: app/src/main/java/uk/colessoft/android/hilllist/domain/ScootXMLHandler.kt
package uk.colessoft.android.hilllist.domain

import org.xml.sax.Attributes
import org.xml.sax.SAXException
import org.xml.sax.helpers.DefaultHandler

import java.util.ArrayList

class ScootXMLHandler : DefaultHandler() {
    private var buff: StringBuffer? = null
    var scootBusinesses = ScootBusinesses()
    private var tempBusiness: Business? = null
    private var buffering = false
    private var tempAddress = ArrayList<String>()

    @Throws(SAXException::class)
    override fun startDocument() {
        // Some sort of setting up work
    }

    @Throws(SAXException::class)
    override fun endDocument() {
        // Some sort of finishing up work
    }

    @Throws(SAXException::class)
    override fun startElement(namespaceURI: String, localName: String,
                              qName: String, atts: Attributes) {
        when (localName) {
            "companyname" -> {
                buff = StringBuffer("")
                buffering = true
            }
            "longitude" -> {
                buff = StringBuffer("")
                buffering = true
            }
            "latitude" -> {
                buff = StringBuffer("")
                buffering = true
            }
            "result" -> {
                buff = StringBuffer("")
                buffering = true
            }
            "telephone" -> {
                buff = StringBuffer("")
                buffering = true
            }
            "postcode" -> {
                buff = StringBuffer("")
                buffering = true
            }
            "linkback" -> {
                buff = StringBuffer("")
                buffering = true
            }
            "l1address" -> {

                buff = StringBuffer("")
                buffering = true
            }
            "l2address" -> {

                buff = StringBuffer("")
                buffering = true
            }
            "l3address" -> {

                buff = StringBuffer("")
                buffering = true
            }
            "l4address" -> {

                buff = StringBuffer("")
                buffering = true
            }
            "l5address" -> {

                buff = StringBuffer("")
                buffering = true
            }
            "l6address" -> {

                buff = StringBuffer("")
                buffering = true
            }
        }
    }

    override fun characters(ch: CharArray, start: Int, length: Int) {
        if (buffering) {
            buff!!.append(ch, start, length)
        }
    }

    @Throws(SAXException::class)
    override fun endElement(namespaceURI: String, localName: String, qName: String) {
        when (localName) {
            "result" -> {
                buffering = false
                tempBusiness = Business()
                tempBusiness!!.resultNumber = Integer.parseInt(buff!!.toString())
                if (this.scootBusinesses.businesses == null)
                    this.scootBusinesses.businesses = ArrayList()
                this.scootBusinesses.businesses!!.add(tempBusiness!!)
            }
            "companyname" -> {
                buffering = false
                tempBusiness!!.companyname = buff!!.toString()
            }
            "longitude" -> {
                buffering = false
                tempBusiness!!.longitude = java.lang.Float.parseFloat(buff!!.toString())
            }
            "latitude" -> {

                buffering = false
                tempBusiness!!.latitude = java.lang.Float.parseFloat(buff!!.toString())
            }
            "telephone" -> {

                buffering = false
                tempBusiness!!.telephone = buff!!.toString()
            }
            "postcode" -> {

                buffering = false
                tempBusiness!!.postCode = buff!!.toString()
            }
            "linkback" -> {

                buffering = false
                tempBusiness!!.scootLink = buff!!.toString()
            }
            "l1address" -> {
                tempAddress = ArrayList()
                buffering = false
                tempAddress.add(buff!!.toString())
            }
            "l2address" -> {
                buffering = false
                tempAddress.add(buff!!.toString())
            }
            "l3address" -> {
                buffering = false
                tempAddress.add(buff!!.toString())
            }
            "l4address" -> {
                buffering = false
                tempAddress.add(buff!!.toString())
            }
            "l5address" -> {
                buffering = false
                tempAddress.add(buff!!.toString())
            }
            "l6address" -> {
                buffering = false
                tempAddress.add(buff!!.toString())
                tempBusiness!!.address = tempAddress
            }
        }// Do something with the full text content that we've just parsed
    }
}